flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_FWC.js

```diff
@@ -57,16 +57,30 @@ class A32NX_FWC {
         // ESDL 1. 0.320
         this.memoLdgInhibit_conf01 = new NXLogic_ConfirmNode(3, true); // CONF 01
 
-        // master warning & caution buttons
+        // Master Warning & Caution
         this.warningPressed = false;
         this.cautionPressed = false;
+        this.warningPriority = false;
 
-        // altitude warning
+        // Altitude Warning
         this.previousTargetAltitude = NaN;
         this._wasBellowThreshold = false;
         this._wasAboveThreshold = false;
         this._wasInRange = false;
         this._wasReach200ft = false;
+
+        // Autopilot Warning
+        this.AutothrottleWarningCanceled = true;
+        this.AutopilotWarningCanceled = true;
+        this.athrdeltaTime = 0;
+        this.apdeltaTime = 0;
+
+        // AUTOPILOT DISCONNECT
+        this.ATHRDisconnectByThrottle = true;
+        this.APDisconnectedBySidestick = true;
+
+        // Update Throttler
+        this.updateThrottler = new UpdateThrottler(200);
```

Comment:
**issue**: This type is doing too many things. The comments above fields really indicate this.
**suggestion**: Split up the type into multiple distinct handlers: one for each type of warning/caution/sound.

Reply:
you mean make other update throttler object and split by warning / caution / sound ?

Reply:
No, the comment is not specific to that line.

The `A32NX_FWC` class has too many fields. It's doing too many things. The `A32NX_FWC` should simply invoke many other types to determine state. E.g. I'd expect an `AThrDisconnect` type which would be invoked by `A32NX_FWC`. Then `AThrDisconnect` would only contain the fields it needs to determine if that warning/caution should be emitted.

Reply:
Okay I understand that!

Reply:
I try that but I can't reduce it more.

2 field is timer, 2 field is status...